### PR TITLE
Update anchor-lang and anchor-spl to the latest version

### PR DESCRIPTION
**Pull request by wuuer**
Update anchor-lang and anchor-spl to the latest version.
Update the latest way to get the seed.
replace "init" with "init_if_needed" for vault field in the struct InitializeVault<'info>.
use "@coral-xyz/anchor" instead of "@project-serum/anchor" for the test typescript. 
delete unnecessary parameters in "accounts()" in the test typescript.

### CODE REVIEW

File: programs/signer-authorization/src/lib.rs [hunk 1]
```diff
@@ -37,7 +35,9 @@ pub mod signer_authorization {
 #[derive(Accounts)]
 pub struct InitializeVault<'info> {
     #[account(
-        init,
+        // We use init_if_needed here for the test. Otherwise, the test will result in an error "Already in use" after testing once.
```
Comment (mikemaccana) [inline programs/signer-authorization/src/lib.rs:38]: Are you sure? The validator started by Anchor tests doesn't keep state. Running the test suite a second time should start a new validator with blank state.

File: tests/signer-authorization.ts [hunk 2]
```diff
@@ -30,35 +22,42 @@ describe("signer-authorization", () => {
...
+        lastValidBlockHeight: latestBlockHash.lastValidBlockHeight,
+        signature: airdropSignature,
+      },
       "confirmed"
-    )
-  })
+    );
+  });
 
   it("Initialize Vault", async () => {
```
Comment (mikemaccana) [inline tests/signer-authorization.ts:51]: This is a nitpick, and was an issue in the old code, but this is a mix of TDD/BDD style.

Either

```
describe('apple')
it('is delicious)

describe('apples')
test('the apple tastes delicious)
```

You might want to use `test()`

File: programs/signer-authorization/src/lib.rs [hunk 3]
```diff
@@ -37,7 +35,9 @@ pub mod signer_authorization {
...
+        // Use "init" if you want to ensure that the "initialize_vault" function runs only once.
+        init_if_needed, 
         payer = authority,
         space = 8 + 32 + 32,
```
Comment (mikemaccana) [inline programs/signer-authorization/src/lib.rs:42]: Use INIT_SPACE. See CONTRIBUTING.md.

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Thanks! A few minor fixes but we can get this in ASAP.

File: programs/signer-authorization/src/lib.rs (hunk 1 above)
Comment (wuuer) [reply programs/signer-authorization/src/lib.rs:40]: The `solana-test-validator` command has an optional parameter:
```
-r, --reset
            Reset the ledger to genesis if it exists. By default the validator will resume an existing ledger (if
            present)
```
If you don't pass this parameter, the previous state will be retained.
I believe beginners might overlook this parameter :)

File: programs/signer-authorization/Cargo.toml [hunk 4]
```diff
@@ -14,7 +14,8 @@ no-idl = []
 no-log-ix-name = []
 cpi = ["no-entrypoint"]
 default = []
+idl-build = ["anchor-lang/idl-build", "anchor-spl/idl-build"]
 
 [dependencies]
-anchor-lang = "0.25.0"
-anchor-spl = "0.25.0"
\ No newline at end of file
+anchor-lang = { version = "0.30.1", features = ["init-if-needed"] }
```
[the remark this answers is not in the document]
Comment (wuuer) [reply programs/signer-authorization/Cargo.toml]: revoke using init_if_needed

File: programs/signer-authorization/src/lib.rs (hunk 3 above)
Comment (wuuer) [reply programs/signer-authorization/src/lib.rs]: commited

**Pull request merged**